libtcg/src/tcg.c

```diff
@@ -859,14 +859,14 @@ const char *tcg_helper_get_name(TCGContext *s, void *func) {
 
 void tcg_register_helper(void *func, const char *name, int param_count, ...) {
     if (!s_additional_helpers) {
-        s_additional_helpers = g_array_new(TRUE, TRUE, sizeof(TCGHelperInfo));
+        s_additional_helpers = g_array_new(TRUE, TRUE, sizeof(TCGHelperInfo*));
     }
 
-    TCGHelperInfo helper;
-    helper.func = func;
-    helper.name = name;
-    helper.flags = dh_callflag(void);
-    helper.sizemask = 0;
+    TCGHelperInfo* helper = (TCGHelperInfo*)malloc(sizeof(TCGHelperInfo));
```

Comment:
Does this need to be freed somewhere?

Reply:
In the original code, I didn't see anywhere removing the element from "s_additional_helpers".

`./s2e/source/s2e$ grep s_additional_helpers -r

libtcg/src/tcg.c:static GArray *s_additional_helpers;
libtcg/src/tcg.c:    if (!s_additional_helpers) {
libtcg/src/tcg.c:        s_additional_helpers = g_array_new(TRUE, TRUE, sizeof(TCGHelperInfo*));
libtcg/src/tcg.c:    g_array_append_val(s_additional_helpers, helper);
libtcg/src/tcg.c:    TCGHelperInfo *h = g_array_index(s_additional_helpers, TCGHelperInfo*, s_additional_helpers->len - 1);
`

Reply:
sounds good, please run the code through clang-format.

Reply:
I added another commit yesterday, please check.